src/core/evaluator.js

```diff
@@ -1604,55 +1616,65 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
                 xobjs = (resources.get('XObject') || Dict.empty);
               }
 
-              var name = args[0].name;
-              if (xobjsCache.key === name) {
-                if (xobjsCache.texts) {
-                  Util.appendToArray(textContent.items, xobjsCache.texts.items);
-                  Util.extendObj(textContent.styles, xobjsCache.texts.styles);
-                }
+              let name = args[0].name;
+              if (name in skipNonFormXObjs) {
                 break;
               }
 
-              var xobj = xobjs.get(name);
+              let xobj = xobjs.get(name);
               if (!xobj) {
                 break;
               }
               assert(isStream(xobj), 'XObject should be a stream');
 
-              var type = xobj.dict.get('Subtype');
+              let type = xobj.dict.get('Subtype');
               assert(isName(type), 'XObject should have a Name subtype');
 
               if (type.name !== 'Form') {
-                xobjsCache.key = name;
-                xobjsCache.texts = null;
+                skipNonFormXObjs[name] = true;
                 break;
               }
 
               // Use a new `StateManager` to prevent incorrect positioning of
               // textItems *after* the Form XObject, since errors in the data
               // can otherwise prevent `restore` operators from being executed.
               // NOTE: This is only an issue when `options.ignoreErrors = true`.
-              var currentState = stateManager.state.clone();
-              var xObjStateManager = new StateManager(currentState);
+              let currentState = stateManager.state.clone();
+              let xObjStateManager = new StateManager(currentState);
 
-              var matrix = xobj.dict.getArray('Matrix');
+              let matrix = xobj.dict.getArray('Matrix');
               if (isArray(matrix) && matrix.length === 6) {
                 xObjStateManager.transform(matrix);
               }
 
+              // Enqueue the `textContent` chunk before parsing the /Form
+              // XObject.
+              enqueueChunk();
+              let sinkWrapper = {
+                enqueueInvoked: false,
```

Comment:
Not sure, but it is mentioned in https://github.com/mozilla/pdf.js/pull/8488#pullrequestreview-42607568, that `skipNonFormXObj` should have `name` of XObject for which enqueue is not called, maybe that's why we need the code here: https://github.com/mozilla/pdf.js/pull/8488#pullrequestreview-42717620
Am i missing something?